fdk.go

```diff
@@ -176,11 +176,11 @@ func doJSONOnce(handler Handler, ctx context.Context, in io.Reader, out io.Write
 	json.NewEncoder(out).Encode(jsonResponse)
 }
 
-func CtxWithDeadline(parent context.Context, fnDeadline string) (context.Context, context.CancelFunc) {
-	ctx, cancel := context.WithCancel(parent)
+func ctxWithDeadline(ctx context.Context, fnDeadline string) (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
go vet would pick up that this cancel function is leaked in one of the control flow paths fwiw (only the child context gets canceled if we find a deadline, not the cancel's). see prev suggestion

Reply:
fixed.